src/main/java/de/zalando/aruha/nakadi/repository/zookeeper/ZookeeperUtils.java

```diff
@@ -0,0 +1,28 @@
+package de.zalando.aruha.nakadi.repository.zookeeper;
+
+import org.apache.curator.framework.recipes.locks.InterProcessLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.Callable;
+
+public class ZookeeperUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ZookeeperUtils.class);
+
+    private ZookeeperUtils() {
+    }
+
+    public static <V> V runLocked(final Callable<V> callable, final InterProcessLock lock) throws Exception {
+        try {
+            lock.acquire();
+            return callable.call();
+        } finally {
+            try {
+                lock.release();
+            } catch (final Exception e) {
+                LOG.warn("Error occurred when releasing ZK lock", e);
```

Comment:
I don't like the idea of eating exception. In this case it means that there was a failure while releasing lock, so probably it will be better to throw this exception instead of returning valid result.